### PR TITLE
Add 'Building' section to collector docs

### PR DESCRIPTION
**Pull request by svrnm**
We discussed this a few times already that we should group some of the "Building..." and add a new section for them. Here it is. I thought about including [Building a custom collector](https://opentelemetry.io/docs/collector/custom-collector/) or not, and decided against inclusion: I think it is a very important topic that needs to be at a higher level, and it is  also different to the other as it is not something that requires you to write code. That's why this section is now called "Building custom components"

I also dropped/added some qualifiers:
* All mention of "custom" as this is now given by the top level
* Renamed "trace receiver" to only "receiver", as this is just an example but could be used for other components as well (I actually used it as an entry point to write a processor)
* Renamed "custom authenticator" to "authenticator extension" to highlight that this is about extensions as well and _might_ be helpful for other extensions.

### CODE REVIEW

File: content/en/docs/collector/building/authenticator-extension.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
-title: Building a custom authenticator
-weight: 30
+title: Building an authenticator extension
```
Comment (jpkrohling) [inline content/en/docs/collector/building/authenticator-extension.md:2]: An authenticator is an extension. In fact, I think the previous title still resonates better with me.

**Review by jpkrohling — APPROVED** [review]
Despite still preferring the previous title, this LGTM.

File: content/en/docs/collector/building/authenticator-extension.md (hunk 1 above)
Comment (svrnm) [reply content/en/docs/collector/building/authenticator-extension.md:2]:
```suggestion
title: Building an authenticator
```

Comment (svrnm) [reply content/en/docs/collector/building/authenticator-extension.md:2]: works for me as well, do you want the "custom" still be there as well?

Comment (jpkrohling) [reply content/en/docs/collector/building/authenticator-extension.md:2]: I have a slight preference for the version with "custom", but they are both fine.

Comment (svrnm) [reply content/en/docs/collector/building/authenticator-extension.md:2]: I will leave "custom" out more because it makes the navbar links a little shorter

**Pull request merged**